client/components/draft_input/components/input_caption.coffee

```diff
@@ -0,0 +1,212 @@
+React = require 'react'
+window.global = window
+window.process = {env: {NODE_ENV: 'development'}}
+
+{ convertToRaw,
+  convertFromHTML,
+  CompositeDecorator,
+  ContentState,
+  Editor,
+  EditorState,
+  RichUtils,
+  Modifier,
+  getVisibleSelectionRect } = require 'draft-js'
+{ stateToHTML } = require 'draft-js-export-html'
+DraftDecorators = require '../decorators.coffee'
+InputUrl = React.createFactory require './input_url.coffee'
+icons = -> require('../icons.jade') arguments...
+{ div, button, p, a, input } = React.DOM
+editor = (props) -> React.createElement Editor, props
+
+decorator = new CompositeDecorator([
+    {
+      strategy: DraftDecorators.findLinkEntities
+      component: DraftDecorators.Link
+    }
+  ])
+
+module.exports = React.createClass
+  displayName: 'DraftInputCaption'
+
+  getInitialState: ->
+    editorState: EditorState.createEmpty()
+    showUrlInput: false
+    urlValue: ''
+    html: null
+    focus: false
+    selectionTarget: null
+
+  componentDidMount: ->
+    if @props.item.caption?.length
+      blocksFromHTML = convertFromHTML(@props.item.caption)
+      state = ContentState.createFromBlockArray(
+        blocksFromHTML.contentBlocks
+        blocksFromHTML.entityMap
+       )
+      @setState editorState: EditorState.createWithContent(state, decorator)
+
+  onChange: (editorState) ->
+    html = @convertToHtml editorState
+    @setState editorState: editorState, html: html
+    @props.item.caption = html
+
+  focus: ->
+    @setState focus: true
+    @refs.editor.focus()
+
+  onBlur: ->
+    @setState focus: false
+
+  onStyleChange: (e) ->
+    e.preventDefault()
+    @onChange RichUtils.toggleInlineStyle(@state.editorState, e.target.className.toUpperCase())
+
+  convertToHtml: (editorState) ->
+    html = stateToHTML editorState.getCurrentContent()
+    html = html.replace(/(\r\n|\n|\r)/gm,'').replace(/<\/p><p>/g, ' ')
```

Comment:
This is for converting pasted content-- this input doesn't accept line breaks, so when html is pasted we have to merge everything into the same <p>